# Case study on: Testing of Hypothesis



A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.

* The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

* The company needs to check whether there is any dependency between the features Region and Manager.

 Help the company to carry out their study with the help of data provided.

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats


In [3]:
data=pd.read_csv("C:\\Users\\user\\Downloads\\Sales_add.csv")
data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [10]:
from scipy.stats import chi2_contingency

contigency= pd.crosstab(data['Region'], data['Manager'])
contigency

# Chi-square test of independence.
c, p, dof, expected = chi2_contingency(contigency)
print('pValue is :',p)

pValue is : 0.5493991051158094


In [8]:
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

# Insights
* There are no null values

* Average sales before investing in digital marketing is $149239.95

* Average sales after investing in digital marketing is $231123.73

# 1. Testing for increase in sales
Hypothesis:
* Null Hypothesis (H0) :- Sales before digital marketing and after digital marketing are the same.

* Alternate Hypothesis (H1 or Ha) :- Sales after digital marketing is greater than the sales before digital marketing .

As the sample size is small(n < 30), we use one tailed t-test to test our hypothesis

In [9]:
import scipy.stats as stats

tscore, pval = stats.ttest_ind(data['Sales_After_digital_add(in $)'],data['Sales_before_digital_add(in $)'],equal_var=True, alternative='greater')
print('pvalue is :', pval)

pvalue is : 1.3071840034523225e-16


# Conclusion :-
Since the p value is less than 0.05 (Alpha value), we reject the Null Hypothesis. In Short, we can say that there is an increase in Sales after implementing digital marketing.

# 2. The company needs to check whether there is any dependency between the features “Region” and “Manager”.
To check whether there is any dependency between the features “Region” and “Manager” we are setting the below two hypothesis :

* Null Hypothesis (H0) : there is no dependency between the features “Region” and “Manager"
* Alternate Hypothesis (H1 or Ha) : there is significant dependency between the features “Region” and “Manager"

# Insight:-
* Since pValue is higher than alpha (0.05) we conclude that we accept the Null hypothesis.

That means,there is no dependency between the features "Region" and "Manager".